src/main/java/Steps/BasePage.java

```diff
@@ -16,4 +24,20 @@ default String getCurrentUrl() {
     default void setMaxWindowResolution() {
         BrowserСhoice.getDriver().manage().window().maximize();
     }
+
+    default <T> T waitUntilProductPageWillBeLoading() {
```

Comment:
My suggestion is to create utils class, for example "WaitUtils" and move all waiters into this class. Probably amount of waiters will be increased and it's not a good practice to have all waits on BasePage